packages/react-components/lib/header-bar.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { createStyles, makeStyles, AppBar, Toolbar } from '@material-ui/core';
+import { TabPanel, TabPanelProps } from './tab-panel';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Badge from '@material-ui/core/Badge';
+
+export interface HeaderBarProps {
+  tabNames: string[];
+  logoPath?: string;
+  tabPanelData?: TabPanelProps[];
+  children?: React.ReactNode;
+}
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      width: '100%',
+    },
+    logo: {
+      maxWidth: 120,
+      opacity: 1,
+    },
+    tabsContainer: {
+      borderRight: '0.25px solid rgba(251, 252, 255, 0.5)',
+      borderLeft: '0.25px solid rgba(251, 252, 255, 0.5)',
+      flexGrow: 2,
+    },
+    toolbar: {
+      textAlign: 'right',
+      flexGrow: -1,
+    },
+  }),
+);
+
+export const HeaderBar = (props: HeaderBarProps): React.ReactElement => {
+  const { logoPath, tabNames, children, tabPanelData } = props;
+  const classes = useStyles();
+
+  const [value, setValue] = React.useState(0);
+
+  const handleChange = (event: React.ChangeEvent<unknown>, newValue: number) => {
+    setValue(newValue);
+  };
+
+  const populateTabs = () => {
+    const tabs = tabNames.map((tabName, index) => (
+      <Tab key={index} label={tabName} value={index} aria-label={`tab-${index}`} />
+    ));
+    return tabs;
+  };
+
+  const populateTabPanels = (data: TabPanelProps[]): JSX.Element[] => {
+    const tabPanels = data.map((item, idx) => {
+      return (
+        <TabPanel key={idx} value={value} index={idx}>
+          {item.children}
```

Comment:
I don't think this works, at least not in the way react code is normally written. It takes in `TabPanelProps[]`, but only uses the `children`, the other props like `value` and `index` are ignored, it may cause a lot of confusion for the consumer.

Instead of doing it this way, you can just let the consumer create it's own `TabPanel`. e.g.

```tsx
const [tabValue, setTabValue] = React.useState(...);
return (
  <>
    <HeaderBar onTabChange={setTabValue} tabNamesAndValues={...}>...</HeaderBar>
    <TabPanel value={tabValue} index={...}>...</TabPanel>
    <TabPanel value={tabValue} index={...}>...</TabPanel>
  </>
)
```

The downside of the above example is that the consumer need to "magically" know that it should create it's own `TabPanel`.

There are some ways to improve upon it, e.g.

```tsx
const [tabValue, setTabValue] = React.useState(...);
return (
  <>
    <HeaderBar>
      <NavigationBar onChange={...}>
        <Tab .../>
        <Tab .../>
        ...
      </NavigationBar>
      <Toolbar>
        ...
      </Toolbar>
    </HeaderBar>
    <TabPanel value={tabValue} index={...}>...</TabPanel>
    <TabPanel value={tabValue} index={...}>...</TabPanel>
  </>
)
```

This would be pretty close to how the material api is designed. With this, the consumer controls both the `Tab` and the `TabPanel`, so it doesn't need any "magical" knowledge. But of course, the downside is that this is pushing more work onto the consumer, so the `NavigationBar` and `Toolbar` (and `HeaderBar` as well I guess) are going to be mostly just css styles "containers". e.g. In this case, the `NavigationBar` would be acting like the `Tabs` component (with our own styling) in material ui. One point to note is that since the homepage/banner is also a tab and that it probably has a different styling, you should create a `BannerTab` component as well.